### PR TITLE
refactor: service tokens

### PR DESCRIPTION
**Pull request by obinns-dd**
# refactor: service tokens

Update to the latest version of the networking package: https://github.com/govuk-one-login/mobile-ios-networking/pull/26
Add the Service Token logic into the One Login app.

Update dependencies following merges of:
- [x] https://github.com/govuk-one-login/mobile-ios-authentication/pull/52
- [x] https://github.com/govuk-one-login/mobile-ios-networking/pull/26
- [x] https://github.com/govuk-one-login/mobile-wallet-ios/pull/178

# Checklist

## Before raising your pull request:
- [x] Commit messages that conform to conventional commit messages
- [x] Ran the app locally ensuring it builds 
- [x] Ran the tests locally ensuring they pass on Build
- [x] Pull request has a clear title with a short description about the feature or update
- [x] Created a `draft` pull request if it is not yet ready for review

## Before your pull request can be reviewed:
- [x] Met all of the acceptance criteria specified in the user story on Jira
- [x] Reviewed your own code to ensure you are following the style guidelines
- [x] Ran the app and tested the feature on a range of device sizes
      Please include iPod Touch, iPhone SE and iPhone 11 as a minimum.
- [x] Written Unit and Integration tests if needed

- [ ] ~~Met all accessibility requirements?~~ - N/A : No user facing change
    - [ ] Checked dynamic type sizes are applied
    - [ ] Checked VoiceOver can navigate your new code
    - [ ] Checked a user can navigate only using a keyboard around your new code 

## Before merging your pull request:
- [x] Ensure that the code coverage and SonarCloud checks have passed
- [x] Actioned and resolved all comments, reaching out to reviewers for clarifications if necessary.
- [x] Ran the app to ensure that no regressions have been caused by changes during code review.

### CODE REVIEW

File: Sources/Tabs/WalletCoordinator.swift [hunk 1]
```diff
@@ -38,7 +38,8 @@ final class WalletCoordinator: NSObject,
                         with: root,
                         networkClient: networkClient,
                         analyticsService: analyticsCenter.analyticsService,
-                        localAuthService: DummyLocalAuthService())
+                        localAuthService: DummyLocalAuthService(),
+                        credentialIssuer: "")
```
Comment (obinns-dd) [inline Sources/Tabs/WalletCoordinator.swift:42]: The requirement to pass in a `credentialIssuer` was a breaking change in the Wallet SDK:
https://github.com/govuk-one-login/mobile-wallet-ios/pull/174

This was not marked as a breaking change when it was merged and the documentation is not clear.
Hopefully @benmckillop can advise us on the correct value for this when he is back from leave,

File: Sources/Utilities/Storage/Session/PersistentSessionManager.swift
Comment (DylanRJ) [inline Sources/Utilities/Storage/Session/PersistentSessionManager.swift:1]: Is there a reason we've changed `accessToken` to `subjectToken`?

**Review by DylanRJ — APPROVED** [review]
🙌

File: Sources/Utilities/Storage/Session/PersistentSessionManager.swift
Comment (obinns-dd) [reply Sources/Utilities/Storage/Session/PersistentSessionManager.swift:1]: Just aligning our terminology with the OAuth spec: https://www.authlete.com/developers/token_exchange/
😊

File: Sources/Tabs/WalletCoordinator.swift (hunk 1 above)
Comment (benmckillop) [reply Sources/Tabs/WalletCoordinator.swift:42]: Resolved this now :) Will update the Wallet readme separately to instruct what to do for this value.

Apologies it wasn't marked as a breaking change - it should have been!

Comment (obinns-dd) [reply Sources/Tabs/WalletCoordinator.swift]: Easily done 🙂 we should look at the automated `diagnose-breaking-api-changes` tool again 👀 
https://x.com/fabianfett/status/1504393754412822530?s=12

File: Sources/Utilities/Storage/Session/Authorization/URLRequest+Authorization.swift [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        urlParser.queryItems = [
+            URLQueryItem(name: "grant_type", value: "urn:ietf:params:oauth:grant-type:token-exchange"),
+            URLQueryItem(name: "scope", value: scope),
+            URLQueryItem(name: "subject_token", value: subjectToken),
+            URLQueryItem(name: "subject_token_type", value: "urn:ietf:params:oauth:token-type:access_token")
+        ]
+        request.httpBody = urlParser.percentEncodedQuery?.data(using: .utf8)
+        return request
+    }
+}
```
Comment (Magdaline-A) [inline Sources/Utilities/Storage/Session/Authorization/URLRequest+Authorization.swift:18]: Is this file the "replacement" of what was removed from the Networking package?

**Review by Magdaline-A — APPROVED** [review]

File: Sources/Utilities/Storage/Session/Authorization/URLRequest+Authorization.swift (hunk 2 above)
Comment (obinns-dd) [reply Sources/Utilities/Storage/Session/Authorization/URLRequest+Authorization.swift:18]: Yes, exactly 🙂 this, plus the implementation of `TokenHolder`

**Review by mcgarj — APPROVED** [review]

**Pull request merged**